python/GafferUI/ColorChooser.py

```diff
@@ -153,42 +160,44 @@ def getDynamicColors( self ) :
 	def _drawBackground( self, painter ) :
 
 		size = self.size()
-		grad = QtGui.QLinearGradient( 0, 0, size.x, 0 )
 
-		displayTransform = self.displayTransform()
+		if self.__gradientToDraw is None :
+			self.__gradientToDraw = QtGui.QLinearGradient( 0, 0, size.x, 0 )
```

Comment:
With the optimization we run into a little trouble if the slider is resized. The gradient doesn't update to account for the new width.

![image](https://github.com/user-attachments/assets/7db4bfa6-0eab-4a7a-968c-54856641ed0b)

Reply:
Right, good catch. I don't see a resize signal on `Widget` to use, so in https://github.com/GafferHQ/gaffer/pull/6111/commits/dd5aad0e71213a37c7cf40b0f0f69616a40999be I just track the size and redraw when it has changed.